enhancements/console/customize-catalogs.md

```diff
@@ -0,0 +1,178 @@
+---
+title: customize-catalogs
+authors:
+  - "@lokanandaprabhu"
+reviewers:
+  - "@jerolimov"
+  - "@spadgett"
+approvers:
+  - "@spadgett"
+  - "@cvogt"
+api-approvers:
+  - "@jerolimov"
+  - "@cvogt"
+creation-date: 2022-07-27
+last-updated: 2022-08-18
+tracking-link: 
+  - https://issues.redhat.com/browse/ODC-6695
+status: implementable
+---
+
+# Customize Catalogs
+
+## Summary
+
+In the developer catalog, there are categories and sub-categories,
+for each category and sub-category there are sub-catalogs/types
+that user can install. This document provides details about how 
+cluster administrator can hide developer catalog and sub-catalogs. 
+
+https://drive.google.com/file/d/1Pn-ZyWk0qBgquVMUhfG8H_BXNN9dWBWy/view?usp=sharing , 
+this is the screenshot for reference, under Developer Catalog,
+there are categories and sub-categories, when user selects, they can see the list
+of types to install (Which is in the bottom left corner). Cluster administrator 
+wants to hide these sub-catalogs.
+
+## Motivation
+
+Cluster Administrator don't want the users to use the catalog 
+to install anything in the cluster until
+the permission is given. Cluster administrator wants to 
+disable developer catalog and sub-catalogs for all users.
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, I want to hide Developer catalog for all users.
+
+#### Story 2
+
+As a cluster admin, I want to hide sub-catalogs for all users.
+
+### Goals
+
+- Allow cluster admins to hide developer catalog and sub-catalogs.
+- Provide information to admin about sub-catalogs they can hide from users.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console CRD`. 
+We can add 
+`spec.customization.developerCatalog.types` which can have list of sub-catalogs disabled.
+
+### Workflow Description
+
+```text
+├── spec
+│   ├── customization
+|       ├── developerCatalog 
+|           ├── types
+|               └──       
+└── ...
+```
+
+Each catalog conforms to the following schema:
+
+// +optional
+- `types: authV1.ResourceAttributes[]` , defines a list of sub-catalogs to hide. 
+Types will vary based on the operator installed. Cluster administrator will be provided with 
+the snippet about the available types.
+
+### API Extensions
+
+Extend the existing `Console` CRD to add `spec.customization.developerCatalog.types` which 
+can have list of sub-catalogs disabled. 
+
+So the example yaml should look like this one:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata: 
+  name: cluster
+  ...
+spec: 
+  customization:
+    developerCatalog:
+      types: 
+        - Devfile
+        - HelmChart
+  ...
+```
+
+If all the available types are listed in `spec.customization.developerCatalog.types` 
+then complete developer catalog will be disabled. A snippet will be provided to the cluster administrator
```

Comment:
We already have customization for categories and sub-categories, there also we are disabling the categories which are mentioned in list and the attribute name is `categories` . So we thought of keeping as `types`. If you suggest to keep it as  `disableTypes`, we can do that.

Reply:
Changes types as disableTypes.

Reply:
Hey @lokanandaprabhu @spadgett, correct that `types` is wrong when it's a deny-list.

I would recommend to change this to `disabledTypes` so that its more aligned with:

1. QuickStarts `disabled`
https://github.com/openshift/api/blob/master/operator/v1/types_console.go#L192
and
2. Add Page `disabledActions`
https://github.com/openshift/api/blob/master/operator/v1/types_console.go#L202